scripts/nbd_client_manager.py

```diff
@@ -155,14 +156,28 @@ def _remove_persistent_connect_info(device):
 
 def connect_nbd(path, exportname):
     """Connects to a free NBD device using nbd-client and returns its path"""
-    _call(['modprobe', 'nbd'])
-    with FILE_LOCK:
-        nbd_device = _find_unused_nbd_device()
-        cmd = ['nbd-client', '-unix', path, nbd_device, '-name', exportname]
-        _call(cmd)
-        _wait_for_nbd_device(nbd_device=nbd_device, connected=True)
-        _persist_connect_info(nbd_device, path, exportname)
-    return nbd_device
+    _call(['modprobe', 'nbd', 'nbds_max=24'])
```

Comment:
Why 24 now, did you mean 1024?

Reply:
No, we've discussed with Stefano, we had to reduce the number due to the memory usage. We're not sure how much they use.

Reply:
Could you add a comment about that? Also do you know how much more code has to be converted over to use NBD directly instead of the block device? 
VDI export should now use NBD directly, how about the emulation (qemu-trad/qemu-upstream), does it connect straight to NBD as well?

Reply:
Good idea, I've added a comment.
Currently nothing uses NBD directly, as far as I know. There are many places in the code that we need to convert, but we haven't started it yet.

Reply:
Not even vhd-tool?

Reply:
No, we have a ticket for it though :disappointed: